### PR TITLE
fix: single select calling onBlur even on scrollbar click

### PR DESCRIPTION
**Pull request by abhishekashyap**
# PR Checklist

## Description
(Replace This Text: Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes and API changes)
- [ ] Build changes
- [ ] CI changes
- [ ] Document content changes
- [ ] Performance improvement
- [ ] Add missing tests
- [ ] Others (please describe)

## How has this been tested?
(Replace This Text: Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.)

[ ] Test A

[ ] Test B

## What is the current behaviour?
(Replace This Text: Please describe the current behaviour you are modifying, or link a relevant issue.)

## What is the new behaviour?
(Replace This Text: Please describe the expected new behaviour.)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

**Note:** (Replace This Text: If this PR contains a breaking change please describe the impact and migration path for existing application.)

## Additional context
(Replace This Text: Please describe any other related information or add screenshots of the PR.)

## Checklist

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [ ] I have commented my code, particularly in hard-to-understand areas

- [ ] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] New and existing unit tests pass locally with my changes

- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/core/src/components/SingleSelect/useKeyboardNavigation.tsx [hunk 1]
```diff
@@ -74,6 +77,13 @@ export const useKeyboardNavigation = (props: Props) => {
             hoverOption(nestedOptions.current.pop()!);
     }, [leftPress, areOptionsVisible]);
 
+    useEffect(() => {
```
Comment (gmukul01) [inline packages/core/src/components/SingleSelect/useKeyboardNavigation.tsx:80]: Could you please write tests for this also ?

File: packages/core/src/components/SingleSelect/SingleSelect.tsx [hunk 2]
```diff
@@ -214,6 +207,7 @@ const Component: FC<SingleSelectProps> = memo(
                         onOptionClick={handleOptionClick}
                         maxWidth={maxWidth}
                         includesNestedOptions={includesNestedOptions}
+                        readOnly
```
Comment (gmukul01) [inline packages/core/src/components/SingleSelect/SingleSelect.tsx:210]: Why did you add `readOnly` ?

**Review by gmukul01 — CHANGES_REQUESTED** [review]

File: packages/core/src/components/SingleSelect/useKeyboardNavigation.tsx (hunk 1 above)
Comment (abhishekashyap) [reply packages/core/src/components/SingleSelect/useKeyboardNavigation.tsx:80]: Added

File: packages/core/src/components/SingleSelect/SingleSelect.tsx (hunk 2 above)
Comment (abhishekashyap) [reply packages/core/src/components/SingleSelect/SingleSelect.tsx]: This was done to not let div get the focus. But since it is now working without it, this can be removed. Thanks

**Review by gmukul01 — APPROVED** [review]
